store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -0,0 +1,218 @@
+
+            // step 5: Copy updatedJsonArray to currentJsonObject
+            currentJsonObject.put("server/stores", currentJsonArray);
+
+            // step 6: Set ldap attribute
+            provisioning.getLocalServer().setServerExternalStoreConfig(currentJsonObject.toString());
+
+        } catch (JSONException e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Add server level external volume properties
+     * @param volumeId
+     * @param server
+     * @throws JSONException, ServiceException
+     */
+    public void addServerProperties(VolumeInfo volInfo) throws ServiceException, JSONException {
+        VolumeExternalInfo volExtInfo = volInfo.getVolumeExternalInfo();
+        try {
+            // step 1: Create and update json object and array for new volume entry
+            JSONObject volExtInfoObj = new JSONObject();
+            volExtInfoObj.put("volumeId", String.valueOf(volInfo.getId()));
+            volExtInfoObj.put("storageType", String.valueOf(volExtInfo.getStorageType()));
+            volExtInfoObj.put("volumePrefix", String.valueOf(volExtInfo.getVolumePrefix()));
+            volExtInfoObj.put("useInFrequentAccess", String.valueOf(volExtInfo.getUseInFrequentAccess()));
+            volExtInfoObj.put("useIntelligentTiering", String.valueOf(volExtInfo.getUseIntelligentTiering()));
+            volExtInfoObj.put("glbBucketConfigId", String.valueOf(volExtInfo.getGlobalBucketConfigurationId()));
+            volExtInfoObj.put("useInFrequentAccessThreshold", String.valueOf(volExtInfo.getUseInFrequentAccessThreshold()));
+
+            // step 2: Fetch current JSON state
+            String serverExternalStoreConfigJson = provisioning.getLocalServer().getServerExternalStoreConfig();
+            JSONObject currentJsonObject = null;
+            if (StringUtil.isNullOrEmpty(serverExternalStoreConfigJson)) {
+                // If current JSON state is already empty, initialise current JSON state Object
+                currentJsonObject = new JSONObject();
+                currentJsonObject.put("server/stores", new JSONArray());
+            } else {
+                // Else convert current JSON state string to current JSON state Object
+                currentJsonObject = new JSONObject(serverExternalStoreConfigJson);
+            }
+
+            // step 3: Fetch current JSON state array
+            JSONArray currentJsonArray = currentJsonObject.getJSONArray("server/stores");
+
+            // step 4: Append current JSON state array with new volume entry
+            currentJsonArray.put(volExtInfoObj);
+
+            // step 5: Overwrite Appended current JSON state array with current JSON state Object
+            currentJsonObject.put("server/stores", currentJsonArray);
+
+            // step 6: Set ldap attribute
+            provisioning.getLocalServer().setServerExternalStoreConfig(currentJsonObject.toString());
+        } catch (JSONException e) {
+            throw e;
```

Comment:
Can add a log before throwing exception.

Reply:
is it really necessary as that exception is caught by caller of this functions and added log over there